### PR TITLE
Update MailSettings for new Laravel mail config structure.

### PR DESCRIPTION
**Pull request by mjauvin**
Requires https://github.com/wintercms/storm/pull/79

Removed the following drivers configs not native in Laravel:
- Mailgun
- Mailchimp Mandrill
- Postmark
- Sparkpost
- Amazon SES

### CODE REVIEW

**Comment by mjauvin** [comment]
I've tested the following drivers:
- php mail
- sendmail
- smtp
- log file

File: modules/system/models/MailSetting.php [hunk 1]
```diff
@@ -85,62 +83,52 @@ public function getSendModeOptions()
...
 
             case self::MODE_SENDMAIL:
-                $config->set('mail.sendmail', $settings->sendmail_path);
+                $config->set('mail.mailers.sendmail.path', $settings->sendmail_path);
                 break;
 
             case self::MODE_MAILGUN:
                 $config->set('services.mailgun.domain', $settings->mailgun_domain);
                 $config->set('services.mailgun.secret', $settings->mailgun_secret);
                 break;
```
Comment (LukeTowers) [inline modules/system/models/MailSetting.php]: If I remember correctly, the mailgun driver also isn't supported by default in Laravel 9 and will require a driver plugin for as well.

Comment (mjauvin) [reply modules/system/models/MailSetting.php]: I see this in Laravel MailManager:

```
    /**
     * Create an instance of the Symfony Mailgun Transport driver.
     *
     * @param  array  $config
     * @return \Symfony\Component\Mailer\Bridge\Mailgun\Transport\MailgunApiTransport
     */
    protected function createMailgunTransport(array $config)
    {   
        $factory = new MailgunTransportFactory();

        if (! isset($config['secret'])) {
            $config = $this->app['config']->get('services.mailgun', []);
        }

        return $factory->create(new Dsn(
            'mailgun+'.($config['scheme'] ?? 'https'),
            $config['endpoint'] ?? 'default',
            $config['secret'],
            $config['domain']
        ));
    }
```

**Comment by mjauvin** [comment]
All done @LukeTowers 

Winter's drivers plugins are also fully functional (but not tested at transport level)

**Comment by mjauvin** [comment]
@LukeTowers I'm pretty confident about those changes and the related https://github.com/wintercms/storm/pull/79

I've done a lot of tests back and forth between the different base drivers (as well as the new Mail driver plugins) and it all works as expected.

Only thing left is to confirm the Sparkpost & Mailgun drivers mapping, but it doesn't prevent this from being merged.
(the fact the other 3 tested driver works means all parts are "speaking to each other" as expected).

**Pull request merged**